### PR TITLE
ISPN-9176 Hot Rod Lightweight Transaction (XaResource)

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-9176

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+         if (trace) {
+            log.tracef("Registering remote cache '%s' for transaction xid=%s", remoteCache.getName(), currentXid);
+         }
+         return new TransactionContext<>(remoteCache.keyMarshaller(), remoteCache.valueMarshaller(),
+               remoteCache.getOperationsFactory(), remoteCache.getName());
+      }
+
+      private void cleanup() {
+         registeredTransactions.remove(transaction);
+         registeredCaches.clear();
```
Comment (rvansa) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java]: Do we need to clear/null the fields? I'd expect that this adapter gets forgotten and gced...

Comment (pruivo) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java]: I think we do. 
Although we don't have recovery in place, the TM can keep this resource for a while if another resource may need some recovery.
Also, (I need to investigate), when recovery is in place, the instance may reincarnate as another transaction and we need to repopulate the `preparedCaches` and `currentXid`.

**Comment by pruivo** [comment]
updated! and no failures!?

**Comment by rvansa** [comment]
Poor CI, almost 2 hours running the tests to no avail... :) Integrating, thanks.

**Comment by danberindei** [comment]
@pruivo do we have any tests w/ servers joining/crashing? We could add some split partition tests, if you're missing test failures ;)

**Comment by danberindei** [comment]
@pruivo Jokes aside, I'd like to see some tests in the client as well. Not at the level we have in embedded, after all lots of embedded PH tests are quarantined because they were unstable, but still I think *some* coverage would be useful.

**Pull request merged**